packages/toolpad-app/src/toolpad/DomLoader.tsx

```diff
@@ -68,8 +75,43 @@ export type DomAction =
       node: appDom.AppDomNode;
     };
 
+const undoStack: appDom.AppDom[] = [];
+const redoStack: appDom.AppDom[] = [];
+
+const updateUndoStack = throttle((dom: appDom.AppDom) => {
+  undoStack.push(dom);
+}, 1000);
```

Comment:
throttle duration could be reduced here if we feel that we are "skipping" certain "save" points too frequently. In general throttle guards against multiple DOM actions happening one after the other. That sequence of actions for the user doesn't really matter, and we want to save only the "final" state as undo point.

Reply:
I'm not sure whether throttling is the right paradigm here. It feels like a workaround for not doing atomic updates to the dom. Subsequent changes to the same property can be detected by comparing two consecutive updates. 

I'm not 100% convinced that we should use a non-deterministic method like throttling to work around this problem. We could argue that this is only an intermediate step towards a better implementation, but I don't really see a way forward towards an implementation that is predictable for the user.

Reply:
As much as I tested it works as expected. If we do find a specific issue with this approach we can look into alternatives or improve existing implementation, but I don't really see the reason of discarding it all together without any alternative in place. 
If/when there is any better implementation, we can do that as a follow up

Reply:
Changing the value of a select comes to mind, or a checkbox. These are easy to fire in rapid succession, but we want them to have separate stack entries

Reply:
I agree, that's why I mentioned that the value could be reduced if we find reasonable duration in which you can physically execute a change

Reply:
Ah, another use-case that requires atomic changes is when we want to show the history of changes. We'd need to be able to be able to attach a label to each undo stack entry that describes it in human language (e.g. "change property 'disabled' to `true`", or "move the `mySelect` component")